### PR TITLE
Bump IREE to 20240226.813 and adapt to API breaks. 

### PR DESCRIPTION
**Pull request by stellaraccident**
This release pulls in https://github.com/openxla/iree/pull/16486 which makes
substantial changes to torch imports:

* Always generates async code (with a special `$async` suffixed entrypoint that the default entrypoint delegates to).
* Internal structure of the generated code is different, invalidating some tests.

### CODE REVIEW

File: core/shark_turbine/dynamo/tensor.py [hunk 1]
```diff
@@ -53,11 +53,7 @@
 
 from ..importers.fx_importer import FxImporter
 
-DEFAULT_COMPILER_FLAGS = (
-    # Enable asynchronous calling convention.
-    "--iree-execution-model=async-external",
```
Comment (raikonenfnu) [inline core/shark_turbine/dynamo/tensor.py:58]: Do we not need iree-execution-model=async-external anymore?

File: core/shark_turbine/dynamo/executor.py [hunk 2]
```diff
@@ -153,7 +153,7 @@ def __init__(
         self,
         user_module: VmModule,
         device_state: DeviceState,
-        entry_name: str = "main",
+        entry_name: str = "main$async",
```
Comment (raikonenfnu) [inline core/shark_turbine/dynamo/executor.py:156]: Python noob Q: what does the `$` sign do? is this doing something similar to `f"main{async}"` ?

**Review by raikonenfnu — COMMENTED** [review]
Nice! couple Qs:

File: core/shark_turbine/dynamo/tensor.py (hunk 1 above)
Comment (stellaraccident) [reply core/shark_turbine/dynamo/tensor.py:58]: No. Async functions are always generated now (with a sync wrapper for default users).

File: core/shark_turbine/dynamo/executor.py (hunk 2 above)
Comment (stellaraccident) [reply core/shark_turbine/dynamo/executor.py:156]: Nothing. IREE always exports an "$async" suffixed function when compiling from torch. This is selecting that variant.

**Comment by stellaraccident** [comment]
Discussed offline. The models tests are OOM'ing and seem to be unstable at head.

**Pull request merged**